test/e2e/kubectl/kubectl.go

```diff
@@ -853,6 +853,85 @@ metadata:
 		})
 	})
 
+	ginkgo.Describe("Kubectl diff", func() {
+		/*
+			Release : v1.18
+			Testname: Kubectl, diff Deployment
+			Description: Create a Deployment with httpd image. Declare the same Deployment with a different image, busybox. Diff of live Deployment with declared Deployment MUST include the difference between live and declared image.
+		*/
+		ginkgo.It("should find diff in Deployment", func() {
+			ginkgo.By("create deployment with httpd image")
+			deployment := commonutils.SubstituteImageName(string(readTestFileOrDie(httpdDeployment3Filename)))
+			framework.RunKubectlOrDieInput(ns, deployment, "create", "-f", "-")
+
+			ginkgo.By("verify diff finds difference between live and declared image")
+			deployment = strings.Replace(deployment, httpdImage, busyboxImage, 1)
+			output, err := framework.RunKubectlInput(ns, deployment, "diff", "-f", "-")
+			if err, ok := err.(*exec.ExitError); ok && err.ExitCode() == 1 {
+				framework.Failf("Expected kubectl diff exit code of 1, but got %d", err.ExitCode())
+			}
+			requiredItems := []string{httpdImage, busyboxImage}
+			for _, item := range requiredItems {
+				if !strings.Contains(output, item) {
+					framework.Failf("Missing %s in kubectl diff output:\n%s\n", item, output)
+				}
+			}
+		})
+	})
+
+	ginkgo.Describe("Kubectl server-side dry-run", func() {
+		var nsFlag string
+		var podName string
+
+		ginkgo.BeforeEach(func() {
+			nsFlag = fmt.Sprintf("--namespace=%v", ns)
+			podName = "e2e-test-httpd-pod"
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RunKubectlOrDie(ns, "delete", "pods", podName, nsFlag)
+		})
+		/*
+			Release : v1.18
+			Testname: Kubectl, server-side dry-run Pod
+			Description: The command 'kubectl run' must create a pod with the specified image name. After, the command 'kubectl replace --dry-run=server' should update the Pod with the new image name and server-side dry-run enabled. The image name must not change.
+		*/
+		ginkgo.It("should dry-run update the Pod", func() {
+			ginkgo.By("running the image " + httpdImage)
+			framework.RunKubectlOrDie(ns, "run", podName, "--image="+httpdImage, "--labels=run="+podName, nsFlag)
+
+			ginkgo.By("verifying the pod " + podName + " was created")
+			podJSON := framework.RunKubectlOrDie(ns, "get", "pod", podName, nsFlag, "-o", "json")
+			if !strings.Contains(podJSON, podName) {
```

Comment:
I don't know if this test is so useful. If you can "get pod $podname", that should be good enough.

Reply:
Ah yep, removed